assets/javascripts/discourse/controllers/event-rsvp.js.es6

```diff
@@ -14,24 +15,33 @@ export default Ember.Controller.extend({
     const type = this.get('type');
     const topic = this.get('model.topic');
 
-    let usernames = topic.get(`event_${type}`);
+    let user_ids = topic.get(`event_${type}`);
 
-    if (!usernames || !usernames.length) return;
+    if (!user_ids || !user_ids.length) return;
 
     let userList = [];
 
-    usernames.forEach((username, index) => {
-      User.findByUsername(username).then((user) => {
-        userList.push(user);
+    ajax('/calendar-events/rsvp/get_usernames', {
```

Comment:
@fzngagan Running multiple ajax calls (get_usernames then findByUsername) here is inefficient. We need to use a single call and return the users as a serialized list. It'll look something like this:

```
ActiveModel::ArraySerializer.new(users, each_serializer: BasicUserSerializer)
```

Reply:
Yes makes a lot of sense. The reason I went with this approach was, findByUsername calls show method from user's controller and does a lot of `access control` checks and there's no method that works similarly but takes `user_id` as a param.

But BasicUserSerializer in itself seems to give away little data so we can probably skip those checks.